### PR TITLE
Changes in adapters in order to store parentID

### PR DESCRIPTION
**Pull request by gzukowski**
**Clickup Task:** 
https://app.clickup.com/t/24308805/VDM-1372

- [ ] Verify and confirm operation (please post a screenshot) 
- [ ] All STH tests pass
- [ ] All [Scramjet Cloud Platform](https://docs.scramjet.org/platform) tests pass
- [ ] Documentation is updated or no changes

### CODE REVIEW

File: packages/host/src/lib/host.ts [hunk 1]
```diff
@@ -713,14 +713,16 @@ export class Host implements IComponent {
 
             const configs = await sequenceAdapter.list();
 
+            sequenceAdapter.logger.pipe(this.logger);
```
Comment (alicja-gruzdz) [inline packages/host/src/lib/host.ts:716]: sequenceAdapter.logger is already piped above, in line 708.

**Review by alicja-gruzdz — COMMENTED** [review]
It looks like it's going in the right direction, thanks :mechanical_arm: 
Please see a couple of my comments.

File: packages/adapters/src/kubernetes-sequence-adapter.ts [hunk 2]
```diff
@@ -24,8 +25,22 @@ import { detectLanguage } from "./utils";
...
+    if (parentId) {
+        sequenceDir = path.join(sequencesRoot, id + "_" + parentId);
+    } else {
+        [id, parentId] = id.split("_");
+        const valid = IDProvider.isValid(id);
+
+        if (valid) {
+            sequenceDir = path.join(sequencesRoot, id + "_" + parentId);
+        } else {
+            sequenceDir = path.join(sequencesRoot, id);
```
[the remark this answers is not in the document]
Comment (gzukowski) [reply packages/adapters/src/kubernetes-sequence-adapter.ts:41]: If the local id is not valid, it means it is in different format and its structure is already a path for the desires directory so there is no need to concate.

File: packages/host/src/lib/host.ts [hunk 3]
```diff
@@ -765,22 +767,28 @@ export class Host implements IComponent {
...
-                this.sequenceStore.set({ id, config, instances: [], name: sequenceName, location: "STH" });
+                this.sequenceStore.set({ id, parent_id: config.parent_id, config, instances: [], name: sequenceName, location: "STH" });
             }
 
             this.logger.trace(`Sequence identified: ${config.id}`);
 
-            // eslint-disable-next-line max-len
-            await this.cpmConnector?.sendSequenceInfo(id, SequenceMessageCode.SEQUENCE_CREATED, config as unknown as GetSequenceResponse);
+            await this.cpmConnector?.sendSequenceInfo(
+                id, // parentId
```
[the remark this answers is not in the document]
Comment (gzukowski) [reply packages/host/src/lib/host.ts:789]: In this case parent_is is already stored in config, which is being sent together with the id

File: packages/host/src/lib/host.ts (hunk 1 above)
Comment (gzukowski) [reply packages/host/src/lib/host.ts:716]: My bad, I made a mistake

File: packages/adapters/src/dockerode-docker-helper.ts [hunk 4]
```diff
@@ -280,7 +282,27 @@ export class DockerodeDockerHelper implements IDockerHelper {
...
     }
-
+    /**
+     * Access to the value of volume's label
+     * @param volumeName Volume name.
+     * @param labelName Label name.
+     * 
+     * @returns Promise which resolves when volume has been removed.
+     */
+    async getLabelValue(volumeName: string, labelName: string): Promise<string | null> {
```
[the remark this answers is not in the document]
Comment (gzukowski) [reply packages/adapters/src/dockerode-docker-helper.ts:292]: You are right, it is a leftover from other solution

**Review by agne6ka — CHANGES_REQUESTED** [review]
Nice work so far 🙌 

There is one extra info in response:
```
    "container": {
                "image": "scramjetorg/runner:0.35.2",
                "maxMem": 512,
                "exposePortsRange": [
                    30000,
                    32767
                ],
                "hostIp": "0.0.0.0"
            },
```

For more info see: https://github.com/scramjetorg/scramjet-cpm-dev/pull/608#pullrequestreview-1671284710

File: packages/host/src/lib/host.ts [hunk 5]
```diff
@@ -918,44 +896,44 @@ export class Host implements IComponent {
...
-            const payload = req.body || ({} as STHRestAPI.StartSequencePayload);
+        const sequenceId = req.params?.id as string;
+        const payload = req.body || ({} as STHRestAPI.StartSequencePayload);
 
-            if (payload.instanceId) {
-                if (!isStartSequenceEndpointPayloadDTO(payload)) {
-                    return { opStatus: ReasonPhrases.UNPROCESSABLE_ENTITY, error: "Invalid Instance id" };
-                }
+        if (payload.instanceId) {
+            if (!IDProvider.isValid(payload.instanceId)) {
```
Comment (patuwwy) [inline packages/host/src/lib/host.ts:909]: will fail tests. actually we allow any 36chars long instance id and tests use not uuid 36 long strings

**Comment by alicja-gruzdz** [comment]
Replaced with https://github.com/scramjetorg/transform-hub/pull/985

**Pull request closed without merging**